packages/video_player/video_player/test/video_player_test.dart

```diff
@@ -957,6 +996,41 @@ void main() {
     });
   });
 
+  test('updates position', () async {
+    final VideoPlayerController controller = VideoPlayerController.networkUrl(
+      _localhostUri,
+      videoPlayerOptions: VideoPlayerOptions(),
+    );
+
+    await controller.initialize();
+
+    const Duration updatesInterval = Duration(milliseconds: 100);
+
+    final List<Duration> positions = <Duration>[];
+    final Completer<void> intervalUpdateCompleter = Completer<void>();
+
+    // Listen for position updates
+    controller.addListener(() {
+      positions.add(controller.value.position);
+      if (positions.length >= 3 && !intervalUpdateCompleter.isCompleted) {
+        intervalUpdateCompleter.complete();
+      }
+    });
+    await controller.play();
+    for (int i = 0; i < 3; i++) {
+      await Future<void>.delayed(updatesInterval);
+      fakeVideoPlayerPlatform._positions[controller.textureId] =
+          Duration(milliseconds: i * updatesInterval.inMilliseconds);
+    }
+
+    // Wait for at least 3 position updates
+    await intervalUpdateCompleter.future.timeout(const Duration(seconds: 5));
```

Comment:
Tests should not include timeouts, per Flutter style. This should just await the future.